release/models/pcep/openconfig-pcep.yang

```diff
@@ -0,0 +1,377 @@
+          using the Delegation mechanism to update LSP parameters in
+          those PCCs that delegate control over their LSPs to the PCE
+          (RFC8051).";
+    reference
+      "RFC8051: Applicability of a Stateful Path Computation Element (PCE)";
+  }
+
+  typedef lsp-control-type {
+    type enumeration {
+      enum PCE_DELEGATED {
+        description
+          "PCC(Router) delegates control over LSP path to PCE (stateful active)";
+      }
+      enum PCC_CONTROLLED {
+        description
+          "PCC(Router) don't allow or revoke control over LSP path to
+          PCE (stateless)";
+      }
+      enum PCC_REPORT_ONLY {
+        description
+          "PCC(Router) only inform to a PCE whenever the state of an LSP changes,
+          but without delegating control (stateful passive)";
+      }
+    }
+    description
+      "The LSP path can be delegated to PCE, locally controled by PCC
+      or only report information state to the PCE.";
+    reference
+      "RFC8231: Path Computation Element Communication Protocol (PCEP)";
+  }
+
+  grouping pcep-servers-top {
+    description
+      "Configuration and state data for establishing a PCEP session with pce servers.";
+
+    container path-computation-servers {
+      description
+        "Defines pce server parameters.";
+	
+      list path-computation-server {
+        key "pce-server-address";
+	
+        description
+          "Configuration and state information for communication with a pce server.";
+
+        leaf pce-server-address {
+          type leafref {
+            path "../config/pce-server-address";
+          }
+          description
+            "IPv4 address of pce server.";
+        }
+
+        container config {
+          description
+            "PCE server configuration.";
+			
+          uses pce-server-config;
+        }
+		
+        container state {
+          config false;
+          description
+            "PCE server state.";
+			
+          uses pce-server-config;
+        }
+		
+	uses pcep-timers-top;
+	uses pcep-authentication-top;
+      }
+    }
+  }
+	
+  grouping pce-server-config {
+    description
+      "Configuration data for establishing a PCEP session with pce servers.";
+
+    leaf pce-server-address {
+      type oc-inet:ip-address;
+      description
+      	"The destination IP address of the PCE server for PCEP service.
+      	Repeat this step to specify multiple PCE servers to provide redundancy.
+      	When multiple candidate servers are specified for a single client,
+      	they are prioritized according to preference.";
+    }
+	
+    leaf pce-server-id {
+      type string;
+      description
+	      "A unique name for the pce server.";
+    }
+
+    leaf pce-server-port {
+      type oc-inet:port-number;
+      default 4189;
+      description
+	      "The destination TCP port used for PCEP service in the PCE server.";
+    }
+
+    leaf pce-client-address {
```

Comment:
Could you check whether we more consistently call this `source-address` in other places? I don't think we refer to the originating address as `client-address` anywhere else, so I would prefer `source-address` I also don't think it makes sense to prepend with `pce-` when it's the PCC's address. Just `source-address` I think is a better fit.

Reply:
> Could you check whether we more consistently call this `source-address` in other places? I don't think we refer to the originating address as `client-address` anywhere else, so I would prefer `source-address` I also don't think it makes sense to prepend with `pce-` when it's the PCC's address. Just `source-address` I think is a better fit.

This is the only comment I have outstanding. I did a quick check, we near universally refer to this as `source-address` everywhere else in the models; unless there's a strong reason to keep the proposed name, I'd like to see this be `source-address` also.

- 18 references to `source-address`
- 1 to `local-source-address` (in telemetry).
- 2 to `source-interface`

Reply:
Changed to source-address as proposed